mods/3d_armor/shields/init.lua

```diff
@@ -53,70 +110,131 @@ if ARMOR_MATERIALS.cactus then
+		reciprocate_damage = true,
+		on_damage = function(player, index, stack)
+			play_sound_effect(player, "default_glass_footstep")
+		end,
+		on_destroy = function(player, index, stack)
+			play_sound_effect(player, "default_break_glass")
+		end,
 	})
 end
 
-if ARMOR_MATERIALS.gold then
-	minetest.register_tool("shields:shield_gold", {
-		description = "Gold Shield",
+if armor.materials.gold then
+	armor:register_armor("shields:shield_gold", {
+		description = S("Gold Shield"),
 		inventory_image = "shields_inv_shield_gold.png",
-		groups = {armor_shield=9, armor_heal=0, armor_use=200},
-		wear = 0,
+		groups = {armor_shield=1, armor_heal=6, armor_use=300,
+			physics_speed=-0.04, physics_gravity=0.04},
+		armor_groups = {fleshy=10},
+		damage_groups = {cracky=1, snappy=2, choppy=2, crumbly=3, level=2},
+		reciprocate_damage = true,
+		on_damage = function(player, index, stack)
+			play_sound_effect(player, "default_dig_metal")
+		end,
+		on_destroy = function(player, index, stack)
+			play_sound_effect(player, "default_dug_metal")
+		end,
 	})
 end
 
-if ARMOR_MATERIALS.mithril then
-	minetest.register_tool("shields:shield_mithril", {
-		description = "Mithril Shield (Warrior)",
+if armor.materials.mithril then
+	armor:register_armor("shields:shield_mithril", {
+		description = S("Mithril Shield (Warrior)"),
 		inventory_image = "shields_inv_shield_mithril.png",
-		groups = {armor_shield=13, armor_heal=0, armor_use=50},
-		wear = 0,
+		groups = {armor_shield=1, armor_heal=12, armor_use=100},
+		armor_groups = {fleshy=15},
+		damage_groups = {cracky=2, snappy=1, level=3},
+		reciprocate_damage = true,
+		on_damage = function(player, index, stack)
+			play_sound_effect(player, "default_glass_footstep")
+		end,
+		on_destroy = function(player, index, stack)
+			play_sound_effect(player, "default_break_glass")
+		end,
 	})
 end
 
-if ARMOR_MATERIALS.blackmithril then
-	minetest.register_tool("shields:shield_blackmithril", {
+if armor.materials.blackmithril then
+	armor:register_armor("shields:shield_blackmithril", {
 		description = "Black Mithril Shield (Warrior)",
 		inventory_image = "shields_inv_shield_black_mithril_warrior.png",
 		groups = {armor_shield=15, armor_heal=0, armor_use=50},
-		wear = 0,
+		armor_groups = {fleshy=12},
+		damage_groups = {cracky=2, snappy=1, level=3},
+		reciprocate_damage = true,
+		--wear = 0,
+		on_damage = function(player, index, stack)
+			play_sound_effect(player, "default_glass_footstep")
+		end,
+		on_destroy = function(player, index, stack)
+			play_sound_effect(player, "default_break_glass")
+		end,
 	})
 end
 
-if ARMOR_MATERIALS.crystal then
-	minetest.register_tool("shields:shield_crystal", {
-		description = "Crystal Shield",
+
+if armor.materials.crystal then
+	armor:register_armor("shields:shield_crystal", {
+		description = S("Crystal Shield"),
 		inventory_image = "shields_inv_shield_crystal.png",
-		groups = {armor_shield=15, armor_heal=0, armor_use=50, armor_fire=1},
```

Comment:
You should probably check with @Darcidride whether or not these stats are specific to MFF.. Modifying tool/armor stats is usually a bad idea around here.

Reply:
I'll un-do those changes.